smac/scenario/scenario.py

```diff
@@ -250,12 +255,24 @@ def extract_instance_specific(instance_list):
                     self.feature_fn)[1]
 
         if self.feature_dict:
-            self.n_features = len(
-                self.feature_dict[list(self.feature_dict.keys())[0]])
             self.feature_array = []
             for inst_ in self.train_insts:
                 self.feature_array.append(self.feature_dict[inst_])
             self.feature_array = numpy.array(self.feature_array)
+            
+            # reduce dimensionality of features of larger than PCA_DIM
+            if self.feature_array.shape[1] > self.PCA_DIM:
+                X = self.feature_array
+                # scale features
```

Comment:
This is a valid argument, but on the other hand we would rely on well-tested code. Inspecting the code again, did you actually check that it works as intended? Calling `X.max()` returns the maximum of the whole array, so this code scales between the maximum of all features and the minimum of all features. This would only cost us a single line, namely the import, or no line if we do `X = MinMaxScaler().fit_transform(X)`.